backend/graminsta/core/services.py

```diff
@@ -32,18 +32,5 @@ def create_or_update_userinfo(validated_data):
             age=validated_data.pop('age'),
             gender=validated_data.pop('gender')
         )[0]
-        return {'user_info': UserInfoSerializer(user_info).data}
+        return {'user_info': user_info}
```

Comment:
I mean, since this function is for creating a `UserInfo` object, you can just return that object directly, instead of putting it in a map. For errors, just throw them instead of putting them in the map too.

The Django REST framework will handle the error and return the corresponding response for you.